compiler/rustc_errors/src/emitter.rs

```diff
@@ -370,7 +370,7 @@ pub trait Emitter: Translate {
     }
 
     fn render_multispan_macro_backtrace(&self, span: &mut MultiSpan, always_backtrace: bool) {
-        let mut new_labels: Vec<(Span, String)> = vec![];
+        let mut new_labels = FxHashSet::default();
```

Comment:
```suggestion
        let mut new_labels = FxIndexSet::default();
```
This will help to avoid `potential_query_instability`.